samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java

```diff
@@ -35,18 +36,26 @@ public boolean canHandle(HandlerInput input) {
         String speechText;
         String favoriteColor = (String) input.getAttributesManager().getSessionAttributes().get(COLOR_KEY);
 
+        boolean isAskResponse = false;
```

Comment:
Seems what this boolean flag does is the opposite of what the variable name means. 

If the response is a "Ask" response, we want to keep the session open (set shouldEndSession to false). L55 is doing the opposite. 

Also, what's the reason for moving this logic from ``MyColorIsIntentHandler`` to here?

Reply:
@tianrenz, My bad. I used the flag in a confusing way as it used to be. 

I should use `boolean noAnswerProvided`, and it works like `        if(noAnswerProvided) {
            responseBuilder.withShouldEndSession(false);
        }`, and guides user to provide a valid color. Will fix in new revision.

Also, I'm not moving `isAskResponse` from `MyColorIsIntentHandler` to `WhatsMyColorIntentHandler`, I'm trying to fix couple issues.

Take a look [here](https://github.com/alexa/alexa-skills-kit-sdk-for-java/blob/2.0.x/samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/MyColorIsIntentHandler.java),`repromptText` never gets return `isAskResponse == false`, `isAskResponse` will be set to `true` when user is not responding correctly. Basically in `MyColorIsIntentHandler `, session should keep **open** since response is asking user to do something.